src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -425,6 +428,11 @@ private static string UrlDecodeInternal(string value, Encoding encoding)
                     helper.AddChar(ch);
             }
```

Comment:
According to the code coverage report, this helper.AddChar(ch) is never executed by tests, which means some of the code paths you added around lazily initializing the array are never touched.  Can you please add tests cases to ensure everything is covered appropriately?